pkg/probe/probes.go

```diff
@@ -73,6 +75,24 @@ func checkApiServerConnectivity(timeout time.Duration) error {
 	})
 }
 
+func checkNodeReadiness(client client.Client, timeout time.Duration) error {
```

Comment:
What if the node was not ready even before we started applying any changes (maybe we applied them to make it ready?). Do you think that may be an issue?

Reply:
This is something that we have to decide and document, then we merge this PR or we don't or maybe we can make this configurable, what do you think ?

Reply:
I think we can just expect the cluster to be Ready before applying anything. If we find out it is an issue for somebody, we can get quorum to discuss it. Sorry, I guess I just wanted to nitpick.